packages/flutter_tools/lib/runner.dart

```diff
@@ -275,9 +271,44 @@ Future<File> _createLocalCrashReport(CrashDetails details) async {
 }
 
 Future<int> _exit(int code, {required ShutdownHooks shutdownHooks}) async {
-  // Prints the welcome message if needed.
+  // Need to get the boolean returned from `messenger.shouldDisplayLicenseTerms()`
+  // before invoking the print welcome method because the print welcome method
+  // will set `messenger.shouldDisplayLicenseTerms()` to false
+  final FirstRunMessenger messenger =
+      FirstRunMessenger(persistentToolState: globals.persistentToolState!);
+  final bool legacyAnalyticsMessageShown =
+      messenger.shouldDisplayLicenseTerms();
+
+  // Prints the welcome message if needed for legacy analytics.
   globals.flutterUsage.printWelcome();
 
+  // Ensure that the consent message has been displayed for unified analytics
+  if (globals.analytics.shouldShowMessage) {
+    globals.logger.printStatus(globals.analytics.getConsentMessage);
+
+    // Because the legacy analytics may have also sent a message,
+    // the conditional below will print additional messaging informing
+    // users that the two consent messages they are receiving is not a
+    // bug
+    if (legacyAnalyticsMessageShown) {
+      globals.logger
+          .printStatus('You have received two consent messages because '
+              'the flutter tool is migrating to a new analytics system. '
+              'Disabling analytics collection will disable both the legacy '
+              'and new analytics collection systems. '
+              'You can disable analytics reporting by running either `flutter --disable-telemetry` '
```

Comment:
`--disable-telemetry` is only for a single run, right? I think we should clarify this.

Reply:
@eliasyishak

Reply:
No, `flutter --disable-telemetry` is the command for completely turning off telemetry for the new package. This message is a follow up to the original PDD consent message

I believe `flutter --suppress-analytics` is for the current run